src/core/packages/saved-objects/api-server-internal/src/lib/search/search_dsl/sorting_params_utils.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import type { SavedObjectsFieldMapping } from '@kbn/core-saved-objects-server';
+import { MappingRuntimeFieldType } from '@elastic/elasticsearch/lib/api/types';
+
+// Priority: numeric > date > keyword > text > others
+const TYPE_PRIORITY = [
+  'long',
+  'integer',
+  'short',
+  'byte',
+  'double',
+  'float',
+  'half_float',
+  'scaled_float',
+  'date',
+  'keyword',
+] as const;
+
+// List of sortable types in Elasticsearch (TYPE_PRIORITY plus 'boolean')
+const SORTABLE_TYPES = new Set([...TYPE_PRIORITY, 'boolean']);
+
+/**
+ * Returns true if the field mapping is sortable in Elasticsearch.
+ * Only keyword, numeric, date, boolean, and similar types are sortable.
+ * Text fields are not sortable.
+ */
+export const isValidSortingField = (fieldMapping?: SavedObjectsFieldMapping): boolean => {
+  if (!fieldMapping) return false;
+  if (fieldMapping.type && SORTABLE_TYPES.has(fieldMapping.type)) return true;
+  // text fields are not sortable, but text with keyword subfield is (via the subfield)
+  return (
+    fieldMapping.type === 'text' &&
+    !!fieldMapping.fields &&
+    Object.values(fieldMapping.fields).some((subField) => subField?.type === 'keyword')
+  );
+};
+
+/**
+ * Returns the name of the keyword subfield if present in a text field mapping.
+ */
+export const getKeywordField = (fieldMapping?: SavedObjectsFieldMapping): string | undefined => {
+  if (fieldMapping?.type === 'text' && fieldMapping.fields) {
+    return Object.entries(fieldMapping.fields).find(
+      ([, subField]) => subField?.type === 'keyword'
+    )?.[0];
+  }
+  return undefined;
+};
+
+/**
+ * Helper to determine the merged type for runtime field.
+ * Maps unsupported runtime types to supported equivalents.
+ */
+export const getMergedFieldType = (
```

Comment:
I didn't get around to testing it, but I couldn't see that we ever check that the two types fields are of the same (or compatible) mappings type? If one field is detected to be of type `long` then we'll use `long` for the runtime mapping. But if another type uses `date` for it's field then we'll get a runtime error when there's a document present with the date field.

This could be bad if someone tests their sorting with just one type of saved object during development, and in production both types are present causing a search error.

Reply:
Done here [`320fae4` (#226324)](https://github.com/elastic/kibana/pull/226324/commits/320fae49b927cf869d56aab6d70421d774a2ea4f)